js/heists-filtering.ts

```diff
@@ -0,0 +1,196 @@
+                const filter = button.getAttribute('data-filter-tab') || 'all';
+                this.setActiveTab(filter);
+                this.filterChallenges();
+            });
+        });
+
+        // Search functionality
+        const searchInput = document.querySelector('[data-search-input]') as HTMLInputElement;
+        if (searchInput) {
+            searchInput.addEventListener('input', (e) => {
+                this.searchTerm = (e.target as HTMLInputElement).value.toLowerCase();
+                this.filterChallenges();
+            });
+        }
+
+        // Difficulty filter
+        const difficultySelect = document.querySelector('[data-difficulty-filter]') as HTMLSelectElement;
+        if (difficultySelect) {
+            difficultySelect.addEventListener('change', () => {
+                this.filterChallenges();
+            });
+        }
+
+        // Time filter
+        const timeSelect = document.querySelector('[data-time-filter]') as HTMLSelectElement;
+        if (timeSelect) {
+            timeSelect.addEventListener('change', () => {
+                this.filterChallenges();
+            });
+        }
+    }
+
+    private setActiveTab(filter: string): void {
+        this.currentFilter = filter;
+        
+        // Update tab button states
+        const tabButtons = document.querySelectorAll('[data-filter-tab]');
+        tabButtons.forEach(button => {
+            const buttonFilter = button.getAttribute('data-filter-tab') || 'all';
+            if (buttonFilter === filter) {
+                button.classList.remove('text-byte-brown-200', 'hover:text-white');
+                button.classList.add('text-white', 'bg-green-900');
+            } else {
+                button.classList.remove('text-white', 'bg-green-900');
+                button.classList.add('text-byte-brown-200', 'hover:text-white');
+            }
+        });
+    }
+
+    private filterChallenges(): void {
+        const challengeCards = document.querySelectorAll('[data-challenge-card]');
+        let visibleCount = 0;
+
+        challengeCards.forEach((card, index) => {
+            const challenge = this.challenges[index];
+            if (!challenge) return;
+
+            const matchesFilter = this.matchesFilter(challenge);
+            const matchesSearch = this.matchesSearch(challenge);
+            const matchesDifficulty = this.matchesDifficulty(challenge);
+            const matchesTime = this.matchesTime(challenge);
+
+            if (matchesFilter && matchesSearch && matchesDifficulty && matchesTime) {
+                card.classList.remove('hidden');
+                visibleCount++;
+            } else {
+                card.classList.add('hidden');
+            }
+        });
+
+        this.updateEmptyState(visibleCount);
+    }
+
+    private matchesFilter(challenge: Challenge): boolean {
+        switch (this.currentFilter) {
+            case 'all':
+                return true;
+            case 'beta':
```

Comment:
Can you create an enum for the challenge categories? The current list is almost certanly going to change so it would be nice if you could change it in one place